code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -134,72 +134,86 @@
 
 /mob/living/carbon/human/adjustCloneLoss(amount)
 	if(dna.species && amount > 0)
-		amount = amount * dna.species.clone_mod
+		amount *= dna.species.clone_mod
 	. = ..()
 
-	var/heal_prob = max(0, 80 - getCloneLoss())
+	if(!amount)
+		return
+
 	var/mut_prob = min(80, getCloneLoss() + 10)
-	if(amount > 0) //cloneloss is being added
-		if(prob(mut_prob))
-			var/list/obj/item/organ/external/candidates = list() //TYPECASTED LISTS ARE NOT A FUCKING THING WHAT THE FUCK
-			for(var/obj/item/organ/external/O in bodyparts)
-				if(O.is_robotic())
-					continue
-				if(!(O.status & ORGAN_MUTATED))
-					candidates |= O
-
-			if(candidates.len)
-				var/obj/item/organ/external/O = pick(candidates)
-				O.mutate()
-				to_chat(src, "<span class='notice'>Something is not right with your [O.name]...</span>")
-				O.add_autopsy_data("Mutation", amount)
-				return
+	var/heal_prob = max(0, 80 - getCloneLoss())
+
+	if(!getCloneLoss()) // All cloneloss was purged - fix all organs
+		for(var/obj/item/organ/external/O in bodyparts)
+			if(O.status & ORGAN_MUTATED)
+				O.unmutate()
+				to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
+		return
 
-	else //cloneloss is being subtracted
-		if(prob(heal_prob))
-			for(var/obj/item/organ/external/O in bodyparts)
-				if(O.status & ORGAN_MUTATED)
-					O.unmutate()
-					to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
-					return
+	if(amount > 0) // Cloneloss was inflicted - chance to mutate an organ
+		if(!prob(mut_prob))
+			return
 
+		var/list/candidates = list()
+		for(var/obj/item/organ/external/O in bodyparts)
+			if(!O.is_robotic() && !(O.status & ORGAN_MUTATED))
+				candidates |= O
+
+		if(length(candidates))
+			var/obj/item/organ/external/O = pick(candidates)
+			O.mutate()
+			to_chat(src, "<span class='notice'>Something is not right with your [O.name]...</span>")
+			O.add_autopsy_data("Mutation", amount)
+	else // Cloneloss was partially healed - chance to unmutate an organ
+		if(!prob(heal_prob))
+			return
 
-	if(getCloneLoss() < 1) //no cloneloss, fixes organs
 		for(var/obj/item/organ/external/O in bodyparts)
 			if(O.status & ORGAN_MUTATED)
 				O.unmutate()
 				to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
+				return
+
+/mob/living/carbon/human/setCloneLoss(amount)
+	if(dna.species && amount > 0)
+		amount *= dna.species.clone_mod
+	. = ..()
 
+	if(!amount) // Cloneloss was set to 0 - fix all organs
+		for(var/obj/item/organ/external/O in bodyparts)
+			if(O.status & ORGAN_MUTATED)
+				O.unmutate()
+				to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
```

Comment:
Can you move this to a separate proc? Since it's used twice now. Best to ensure that the adjust and set have the same flow for curing stuff etc.